File: documentation/user/en/query/filtering/examples/references/entity-having.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		ReferenceHaving(
        			"brand",
        			EntityHaving(
        				AttributeEquals("code", "apple")
        			)
        		)
        	),
        	Require(
        		EntityFetch(
        			AttributeContent("code"),
        			ReferenceContent(
        				"brand",
        				EntityFetch(
        					AttributeContent("code")
        				)
        			)
        		)
        	)
        )
	)
);

File: documentation/user/en/query/filtering/examples/references/entity-having.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					referenceHaving(
						"brand",
						entityHaving(
							attributeEquals("code", "apple")
						)
					)
				),
				require(
					entityFetch(
						attributeContent("code"),
						referenceContent(
							"brand",
							entityFetch(
								attributeContent("code")
							)
						)
					)
				)
			)
		);
	}
);